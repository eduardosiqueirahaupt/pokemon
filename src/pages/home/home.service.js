import axios from "axios";
const initialUrl = 'https://pokeapi.co/api/v2/pokemon/?offset=0&limit=20'
const byIdPokemon = 'https://pokeapi.co/api/v2/pokemon/'
export const allPokemonUrl = 'https://pokeapi.co/api/v2/pokemon?limit=100000&offset=0'
export const fetchData = (url = initialUrl) => (axios.get(url).then(x => x.data))
export const fetchAllById = (ids) => {
    return axios.all(ids.map((id) => axios.get(byIdPokemon + id).then(x => x.data)))
}