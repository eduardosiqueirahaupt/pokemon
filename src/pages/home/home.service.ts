import { IFetchPokemons } from ".";
import { useFetch } from "../../hooks/useFetch";
export interface IFetch {
  data: IFetchPokemons;
}
export const usePokemonsGetData = (): IFetch =>
  useFetch("https://pokeapi.co/api/v2/pokemon/?offset=0&limit=20");
