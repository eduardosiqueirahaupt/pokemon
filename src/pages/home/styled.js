import styled from 'styled-components'

export const Wrapper = styled.div`
  height: calc(100% - 50px);
  position: relative;
`

export const Content = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`

export const WrapperCards = styled.div`
  width: 100%;
`

export const ContentCards = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, 150px);
  grid-row-gap: 12px;
  grid-column-gap: 28px;
  justify-content: center;
`