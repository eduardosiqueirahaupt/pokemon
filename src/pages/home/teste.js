// import { Snackbar } from "@mui/material";
// import { useCallback, useEffect, useState } from "react";
// import Cards from "../../components/cards";
// import CardsFavorites from "../../components/cardsFavorites";
// import Header from "../../components/header-cards";
// import { fetchData } from "./home.service";
// import * as S from "./styled";

// interface IHome {
//   isFavorites: boolean;
// }

// interface Ipokemon {
//   name: string;
// }

// const initialValueFeedback = {
//   isOpen: false,
//   message: "",
// };

// function Home({ isFavorites }: IHome) {
//   const [search, setSearch] = useState("");
//   const [pokeData, setPokeData] = useState<Ipokemon[]>([]);
//   const [feedback, setFeedback] = useState(initialValueFeedback);

//   const handleFeedback = (message: string) => {
//     setFeedback({
//       isOpen: true,
//       message: message,
//     });
//   };

//   const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
//     setSearch(e.target.value);
//   };

//   const getData = useCallback(async () => {
//     const _data = await fetchData();
//     setPokeData(_data.results);
//   }, []);

//   const filterPokemons = (): Ipokemon[] =>
//     pokeData.filter((x) => x.name.includes(search.toLowerCase()));

//   useEffect(() => {
//     getData();
//   }, []);

//   return (
//     pokeData && (
//       <S.Wrapper>
//         <Header handleSearch={handleSearch} search={search} />
//         <S.Content>
//           <S.WrapperCards>
//             <S.ContentCards>
//               {isFavorites ? (
//                 <CardsFavorites
//                   data={filterPokemons()}
//                   handleFeedback={handleFeedback}
//                 />
//               ) : (
//                 <Cards
//                   data={filterPokemons()}
//                   handleFeedback={handleFeedback}
//                 />
//               )}
//             </S.ContentCards>
//           </S.WrapperCards>
//         </S.Content>
//         <Snackbar
//           open={feedback.isOpen}
//           autoHideDuration={1000}
//           onClose={() => setFeedback(initialValueFeedback)}
//           message={feedback.message}
//         />
//       </S.Wrapper>
//     )
//   );
// }

// export default Home;

export default () => {
  return <div>
      testes
  </div>;
};
