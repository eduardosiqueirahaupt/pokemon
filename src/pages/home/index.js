import { Snackbar } from '@mui/material';
import { useCallback, useEffect, useState } from 'react';
import Cards from '../../components/cards';
import Header from '../../components/header-cards';
import Pagination from '../../components/pagination';
import Wrapper from '../../components/wrapper';
import { fetchData } from './home.service';

const initialValueFeedback = {
  isOpen: false,
  message: ''
}

const Home = () => {
  const [search, setSearch] = useState('');
  const [pokeData, setPokeData] = useState([]);
  const [feedback, setFeedback] = useState(initialValueFeedback)
  const [pagination, setPagination] = useState({
    next: '',
    previous: ''
  })

  const handleFeedback = (message) => {
    setFeedback({
      isOpen: true,
      message: message
    })
  }

  const handleSearch = (e) => {
    setSearch(e.target.value)
  }

  const filterPokemons = () => pokeData.
    filter(x => x.name.includes(search.toLowerCase()))

  const getData = useCallback(async (url) => {
    const _data = await fetchData(url);
    setPokeData(_data.results)
    setPagination({
      next: _data.next,
      previous: _data.previous
    })
  }, [])

  useEffect(() => {
    getData()
  }, [getData])

  return (
    <>
      <Header
        handleSearch={handleSearch}
        search={search}
      />
      <Wrapper>
        <Cards
          data={filterPokemons()}
          handleFeedback={handleFeedback}
        />
      </Wrapper>
      <Pagination
        previous={() => getData(pagination.previous)}
        next={() => getData(pagination.next)}
        pagination={pagination}
      />
      <Snackbar
        open={feedback.isOpen}
        autoHideDuration={1000}
        onClose={() => setFeedback(initialValueFeedback)}
        message={feedback.message}
      />
    </>
  )
}

export default Home;