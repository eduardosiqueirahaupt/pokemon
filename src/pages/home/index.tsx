import { useState } from "react";
import Cards from "../../components/cards";
import Header from "../../components/header-cards";
import { usePokemonsGetData } from "./home.service";
import * as S from "./styled";

interface Ipokemon {
  name: string;
}

export interface IFetchPokemons {
  results: Ipokemon[] | undefined;
}

function Home() {
  const [search, setSearch] = useState("");
  const { data } = usePokemonsGetData();

  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearch(e.target.value);
  };

  const filterPokemons = () =>
    data?.results?.filter((x) => x.name.includes(search.toLowerCase()));

  if (!data?.results) {
    return <span>Carregando....</span>;
  }

  return (
    <S.Wrapper>
      <Header handleSearch={handleSearch} search={search} />
      <S.Content>
        <S.WrapperCards>
          <S.ContentCards>
            <Cards data={filterPokemons()} />
          </S.ContentCards>
        </S.WrapperCards>
      </S.Content>
    </S.Wrapper>
  );
}

export default Home;
