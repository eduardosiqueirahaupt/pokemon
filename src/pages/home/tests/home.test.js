import { cleanup, fireEvent, render, screen, waitFor, act } from '@testing-library/react'
import Home from '..'
import * as service from '../home.service'

const renderComponent = () => (
  render(
    <Home />
  )
)

describe("<Home/>", () => {

  beforeAll(() => {
    localStorage.setItem("FavPokemons", JSON.stringify([]))
  })

  beforeEach(() => {
    jest.spyOn(service, 'fetchData').mockImplementation(jest.fn(() => Promise.resolve(getPokemonsMock)));
    cleanup()
  })

  it("Deve renderizar com sucesso", async () => {
    const { container } = renderComponent();
    await waitFor(() => screen.findByText(/bulbasaur/))
    expect(container).toMatchSnapshot();
  });

  it("Deve achar pokemon ao carregar tela", async () => {
    renderComponent();
    expect(await waitFor(() => screen.findByText(/bulbasaur/))).toBeDefined();
  })
})

const getPokemonsMock = {
  "count": 1126,
  "next": "https://pokeapi.co/api/v2/pokemon/?offset=20&limit=20",
  "previous": null,
  "results": [
    { "name": "bulbasaur", "url": "https://pokeapi.co/api/v2/pokemon/1/" },
    { "name": "ivysaur", "url": "https://pokeapi.co/api/v2/pokemon/2/" },
    { "name": "venusaur", "url": "https://pokeapi.co/api/v2/pokemon/3/" },
    { "name": "charmander", "url": "https://pokeapi.co/api/v2/pokemon/4/" },
    { "name": "charmeleon", "url": "https://pokeapi.co/api/v2/pokemon/5/" },
    { "name": "charizard", "url": "https://pokeapi.co/api/v2/pokemon/6/" },
    { "name": "squirtle", "url": "https://pokeapi.co/api/v2/pokemon/7/" },
    { "name": "wartortle", "url": "https://pokeapi.co/api/v2/pokemon/8/" },
    { "name": "blastoise", "url": "https://pokeapi.co/api/v2/pokemon/9/" },
    { "name": "caterpie", "url": "https://pokeapi.co/api/v2/pokemon/10/" },
    { "name": "metapod", "url": "https://pokeapi.co/api/v2/pokemon/11/" },
    { "name": "butterfree", "url": "https://pokeapi.co/api/v2/pokemon/12/" },
    { "name": "weedle", "url": "https://pokeapi.co/api/v2/pokemon/13/" },
    { "name": "kakuna", "url": "https://pokeapi.co/api/v2/pokemon/14/" },
    { "name": "beedrill", "url": "https://pokeapi.co/api/v2/pokemon/15/" },
    { "name": "pidgey", "url": "https://pokeapi.co/api/v2/pokemon/16/" },
    { "name": "pidgeotto", "url": "https://pokeapi.co/api/v2/pokemon/17/" },
    { "name": "pidgeot", "url": "https://pokeapi.co/api/v2/pokemon/18/" },
    { "name": "rattata", "url": "https://pokeapi.co/api/v2/pokemon/19/" },
    { "name": "raticate", "url": "https://pokeapi.co/api/v2/pokemon/20/" }
  ]
}
