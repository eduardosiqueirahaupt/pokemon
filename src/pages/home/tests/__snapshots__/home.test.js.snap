// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Deve renderizar com sucesso 1`] = `
<div>
  <div
    class="sc-eCYdqJ QSBIM"
  >
    <input
      class="sc-jSMfEi hBkAaQ"
      data-testid="input-filter"
      name="search"
      placeholder="Buscar..."
      type="text"
      value=""
    />
  </div>
  <div
    class="sc-ftvSup hFhTFb"
  >
    <div
      class="sc-papXJ cYtZtE"
    >
      <div
        class="sc-jqUVSM jjphog"
      >
        <div
          class="sc-kDDrLX kLgwMV"
        >
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              bulbasaur
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-bulbasaur"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              ivysaur
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/2.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-ivysaur"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              venusaur
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/3.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-venusaur"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              charmander
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/4.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-charmander"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              charmeleon
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/5.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-charmeleon"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              charizard
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/6.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-charizard"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              squirtle
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/7.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-squirtle"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              wartortle
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/8.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-wartortle"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              blastoise
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/9.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-blastoise"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              caterpie
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/10.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-caterpie"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              metapod
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/11.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-metapod"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              butterfree
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/12.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-butterfree"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              weedle
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/13.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-weedle"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              kakuna
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/14.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-kakuna"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              beedrill
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/15.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-beedrill"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              pidgey
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/16.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-pidgey"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              pidgeotto
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/17.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-pidgeotto"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              pidgeot
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/18.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-pidgeot"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              rattata
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/19.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-rattata"
              src="add-fav.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              raticate
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/20.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-raticate"
              src="add-fav.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-gKXOVf goYihl"
  >
    <span
      class="sc-iBkjds gLAUHr disabled"
      title="Página Anterior"
    >
      &lt;
    </span>
    <span
      class="sc-iBkjds gLAUHr"
      title="Próxima Página"
    >
      &gt;
    </span>
  </div>
</div>
`;
